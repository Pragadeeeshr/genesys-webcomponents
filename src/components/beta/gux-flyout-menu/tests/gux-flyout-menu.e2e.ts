import { newE2EPage } from '@stencil/core/testing';
import '../../../../components';

describe('gux-flyout-menu', () => {
  const html = `
    <gux-flyout-menu-beta>
      <span slot="target">Example Target Element</span>
      <gux-menu slot="menu">
        <gux-menu-option>Option One</gux-menu-option>
        <gux-submenu label="Submenu Two">
          <gux-menu-option>Option One</gux-menu-option>
          <gux-submenu label="Submenu Two">
            <gux-menu-option>Option One</gux-menu-option>
            <gux-menu-option>Option Two</gux-menu-option>
            <gux-menu-option>Option Three</gux-menu-option>
          </gux-submenu>
          <gux-menu-option>Option Three</gux-menu-option>
        </gux-submenu>
        <gux-menu-option>Option Three</gux-menu-option>
        <gux-menu-option>Option Four</gux-menu-option>
        <gux-submenu label="Submenu Five">
          <gux-menu-option>Option One</gux-menu-option>
          <gux-menu-option>Option Two</gux-menu-option>
          <gux-submenu label="Submenu Three">
            <gux-menu-option>Option One</gux-menu-option>
            <gux-submenu label="Submenu Two">
              <gux-menu-option>Option One</gux-menu-option>
              <gux-submenu label="Submenu Two">
                <gux-menu-option>Option One</gux-menu-option>
                <gux-menu-option>Option Two</gux-menu-option>
                <gux-menu-option>Option Three</gux-menu-option>
              </gux-submenu>
              <gux-menu-option>Option Three</gux-menu-option>
            </gux-submenu>
            <gux-menu-option>Option Three</gux-menu-option>
          </gux-submenu>
        </gux-submenu>
      </gux-menu>
    </gux-flyout-menu-beta>
  `;

  describe('#render', () => {
    it(`should render as expected`, async () => {
      const page = await newE2EPage({ html });
      const element = await page.find('gux-flyout-menu-beta');

      expect(element.outerHTML).toMatchSnapshot();
    });
  });
});
