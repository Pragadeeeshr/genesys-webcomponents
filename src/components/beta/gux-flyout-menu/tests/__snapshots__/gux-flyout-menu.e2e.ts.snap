// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-flyout-menu #render should render as expected 1`] = `"<gux-flyout-menu-beta tabindex=\\"1\\" class=\\"hydrated\\"><!----> <span aria-haspopup=\\"true\\"><span slot=\\"target\\">Example Target Element</span></span><div class=\\"gux-flyout-menu-wrapper\\" data-popper-placement=\\"bottom-start\\" style=\\"position: absolute; left: 0px; top: 0px; margin: 0px; right: auto; bottom: auto; transform: translate(8px, 32px);\\"><div class=\\"gux-flyout-menu-content\\"><gux-menu slot=\\"menu\\" role=\\"menu\\" class=\\"hydrated\\"><!----> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option One</span></button></gux-menu-option> <gux-submenu label=\\"Submenu Two\\" class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-submenu-button\\" tabindex=\\"-1\\" aria-haspopup=\\"true\\"><span class=\\"gux-submenu-button-text\\">Submenu Two</span><gux-icon class=\\"gux-submenu-open-icon hydrated\\" icon-name=\\"chevron-right\\" screenreader-text=\\"Open submenu\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" data-name=\\"Layer 1\\" viewBox=\\"0 0 32 32\\" aria-label=\\"Open submenu\\"> <path d=\\"M25.29 16.84L11.48 30.65a1.17 1.17 0 01-1.67 0l-3.1-3.09a1.17 1.17 0 010-1.68L16.6 16 6.71 6.12a1.17 1.17 0 010-1.67L9.8 1.36a1.16 1.16 0 011.67 0l13.82 13.8a1.17 1.17 0 010 1.67z\\"></path> </svg></div></gux-icon></button><div class=\\"gux-submenu-wrapper\\" data-popper-reference-hidden data-popper-escaped data-popper-placement=\\"right-start\\" style=\\"position: absolute; left: 0px; top: 0px; margin: 0px; right: auto; bottom: auto; transform: translate(0px, 0px);\\"><div role=\\"menu\\" class=\\"gux-submenu-content\\"> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option One</span></button></gux-menu-option> <gux-submenu label=\\"Submenu Two\\" class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-submenu-button\\" tabindex=\\"-1\\" aria-haspopup=\\"true\\"><span class=\\"gux-submenu-button-text\\">Submenu Two</span><gux-icon class=\\"gux-submenu-open-icon hydrated\\" icon-name=\\"chevron-right\\" screenreader-text=\\"Open submenu\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" data-name=\\"Layer 1\\" viewBox=\\"0 0 32 32\\" aria-label=\\"Open submenu\\"> <path d=\\"M25.29 16.84L11.48 30.65a1.17 1.17 0 01-1.67 0l-3.1-3.09a1.17 1.17 0 010-1.68L16.6 16 6.71 6.12a1.17 1.17 0 010-1.67L9.8 1.36a1.16 1.16 0 011.67 0l13.82 13.8a1.17 1.17 0 010 1.67z\\"></path> </svg></div></gux-icon></button><div class=\\"gux-submenu-wrapper\\" data-popper-reference-hidden data-popper-escaped data-popper-placement=\\"right-start\\" style=\\"position: absolute; left: 0px; top: 0px; margin: 0px; right: auto; bottom: auto; transform: translate(0px, 0px);\\"><div role=\\"menu\\" class=\\"gux-submenu-content\\"> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option One</span></button></gux-menu-option> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option Two</span></button></gux-menu-option> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option Three</span></button></gux-menu-option> </div></div></gux-submenu> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option Three</span></button></gux-menu-option> </div></div></gux-submenu> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option Three</span></button></gux-menu-option> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option Four</span></button></gux-menu-option> <gux-submenu label=\\"Submenu Five\\" class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-submenu-button\\" tabindex=\\"-1\\" aria-haspopup=\\"true\\"><span class=\\"gux-submenu-button-text\\">Submenu Five</span><gux-icon class=\\"gux-submenu-open-icon hydrated\\" icon-name=\\"chevron-right\\" screenreader-text=\\"Open submenu\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" data-name=\\"Layer 1\\" viewBox=\\"0 0 32 32\\" aria-label=\\"Open submenu\\"> <path d=\\"M25.29 16.84L11.48 30.65a1.17 1.17 0 01-1.67 0l-3.1-3.09a1.17 1.17 0 010-1.68L16.6 16 6.71 6.12a1.17 1.17 0 010-1.67L9.8 1.36a1.16 1.16 0 011.67 0l13.82 13.8a1.17 1.17 0 010 1.67z\\"></path> </svg></div></gux-icon></button><div class=\\"gux-submenu-wrapper\\" data-popper-reference-hidden data-popper-escaped data-popper-placement=\\"right-start\\" style=\\"position: absolute; left: 0px; top: 0px; margin: 0px; right: auto; bottom: auto; transform: translate(0px, 0px);\\"><div role=\\"menu\\" class=\\"gux-submenu-content\\"> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option One</span></button></gux-menu-option> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option Two</span></button></gux-menu-option> <gux-submenu label=\\"Submenu Three\\" class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-submenu-button\\" tabindex=\\"-1\\" aria-haspopup=\\"true\\"><span class=\\"gux-submenu-button-text\\">Submenu Three</span><gux-icon class=\\"gux-submenu-open-icon hydrated\\" icon-name=\\"chevron-right\\" screenreader-text=\\"Open submenu\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" data-name=\\"Layer 1\\" viewBox=\\"0 0 32 32\\" aria-label=\\"Open submenu\\"> <path d=\\"M25.29 16.84L11.48 30.65a1.17 1.17 0 01-1.67 0l-3.1-3.09a1.17 1.17 0 010-1.68L16.6 16 6.71 6.12a1.17 1.17 0 010-1.67L9.8 1.36a1.16 1.16 0 011.67 0l13.82 13.8a1.17 1.17 0 010 1.67z\\"></path> </svg></div></gux-icon></button><div class=\\"gux-submenu-wrapper\\" data-popper-reference-hidden data-popper-escaped data-popper-placement=\\"right-start\\" style=\\"position: absolute; left: 0px; top: 0px; margin: 0px; right: auto; bottom: auto; transform: translate(0px, 0px);\\"><div role=\\"menu\\" class=\\"gux-submenu-content\\"> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option One</span></button></gux-menu-option> <gux-submenu label=\\"Submenu Two\\" class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-submenu-button\\" tabindex=\\"-1\\" aria-haspopup=\\"true\\"><span class=\\"gux-submenu-button-text\\">Submenu Two</span><gux-icon class=\\"gux-submenu-open-icon hydrated\\" icon-name=\\"chevron-right\\" screenreader-text=\\"Open submenu\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" data-name=\\"Layer 1\\" viewBox=\\"0 0 32 32\\" aria-label=\\"Open submenu\\"> <path d=\\"M25.29 16.84L11.48 30.65a1.17 1.17 0 01-1.67 0l-3.1-3.09a1.17 1.17 0 010-1.68L16.6 16 6.71 6.12a1.17 1.17 0 010-1.67L9.8 1.36a1.16 1.16 0 011.67 0l13.82 13.8a1.17 1.17 0 010 1.67z\\"></path> </svg></div></gux-icon></button><div class=\\"gux-submenu-wrapper\\" data-popper-reference-hidden data-popper-escaped data-popper-placement=\\"right-start\\" style=\\"position: absolute; left: 0px; top: 0px; margin: 0px; right: auto; bottom: auto; transform: translate(0px, 0px);\\"><div role=\\"menu\\" class=\\"gux-submenu-content\\"> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option One</span></button></gux-menu-option> <gux-submenu label=\\"Submenu Two\\" class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-submenu-button\\" tabindex=\\"-1\\" aria-haspopup=\\"true\\"><span class=\\"gux-submenu-button-text\\">Submenu Two</span><gux-icon class=\\"gux-submenu-open-icon hydrated\\" icon-name=\\"chevron-right\\" screenreader-text=\\"Open submenu\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" data-name=\\"Layer 1\\" viewBox=\\"0 0 32 32\\" aria-label=\\"Open submenu\\"> <path d=\\"M25.29 16.84L11.48 30.65a1.17 1.17 0 01-1.67 0l-3.1-3.09a1.17 1.17 0 010-1.68L16.6 16 6.71 6.12a1.17 1.17 0 010-1.67L9.8 1.36a1.16 1.16 0 011.67 0l13.82 13.8a1.17 1.17 0 010 1.67z\\"></path> </svg></div></gux-icon></button><div class=\\"gux-submenu-wrapper\\" data-popper-reference-hidden data-popper-escaped data-popper-placement=\\"right-start\\" style=\\"position: absolute; left: 0px; top: 0px; margin: 0px; right: auto; bottom: auto; transform: translate(0px, 0px);\\"><div role=\\"menu\\" class=\\"gux-submenu-content\\"> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option One</span></button></gux-menu-option> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option Two</span></button></gux-menu-option> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option Three</span></button></gux-menu-option> </div></div></gux-submenu> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option Three</span></button></gux-menu-option> </div></div></gux-submenu> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option Three</span></button></gux-menu-option> </div></div></gux-submenu> </div></div></gux-submenu> </gux-menu></div><div class=\\"gux-arrow\\" data-popper-arrow style=\\"position: absolute; left: 0px; transform: translate(16px, 0px);\\"></div></div></gux-flyout-menu-beta>"`;
